webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js

```diff
@@ -579,53 +595,134 @@ export class SakaiGrader extends gradableDataMixin(SakaiElement) {
   }
 
   toggleFeedback() {
-
     const feedbackPanel = $("#feedback-panel");
 
     if (!feedbackPanel.dialog("instance")) {
       feedbackPanel.dialog({
         width: "auto",
-        close: () => this.toggleFeedback()
+        beforeClose: () => {
+          const isUnsaved = this.isFeedbackUnsaved();
+          if (isUnsaved) {
+            this.cancelFeedbackToogle();
+          }
+          return !isUnsaved;
+        },
+        close: () => this.cancelFeedbackToogle()
       });
       this.feedbackCommentEditor = this.replaceWithEditor("grader-feedback-comment");
-    } else {
-      this.submission.feedbackComment = this.feedbackCommentEditor.getData();
+      this.previousFeedback = this.submission.feedbackComment;
+    }
+    else if (this.isFeedbackUnsaved()) {
+      this.toggleConfirm("confirm-feedback-panel");
+    }
+    else {
       this.feedbackCommentEditor.destroy();
       feedbackPanel.dialog("destroy");
       this.requestUpdate();
     }
   }
 
   doneWithFeedbackDialog() {
-
+    this.submission.feedbackComment = this.feedbackCommentEditor.getData();
+    this.previousFeedback = this.submission.feedbackComment;
+    this.modified = true;
     this.toggleFeedback();
     document.getElementById("grader-feedback-button").focus();
   }
 
-  togglePrivateNotes() {
+  isFeedbackUnsaved() {
+    return this.feedbackCommentEditor.getData() !== this.previousFeedback;
+  }
 
+  isPvtNotesUnsaved() {
```

Comment:
Look at the modified flag usage that's already there. If you want to indicate that it is the private notes that has changed, maybe change the colour of the private notes button, or add an alert icon after it. Actually, the icon is better for a11y.